src/state_machine.zig

```diff
@@ -692,34 +692,59 @@ pub fn StateMachineType(comptime Storage: type, comptime constants_: struct {
             // If the transfer already exists, then it must not influence the overflow or limit checks.
             if (self.get_transfer(t.id)) |e| return create_transfer_exists(t, e);
 
+            const amount = amount: {
+                var amount: u64 = t.amount;
+                if (t.flags.debits_at_most) {
+                    const dr_limit = if (dr.flags.debits_must_not_exceed_credits)
+                        dr.credits_posted
+                    else
+                        std.math.maxInt(u64);
+                    const dr_available = dr_limit - dr.debits_posted - dr.debits_pending;
+                    if (dr_available < amount) amount = dr_available;
+                }
+
+                if (t.flags.credits_at_most) {
+                    const cr_limit = if (cr.flags.credits_must_not_exceed_debits)
+                        cr.debits_posted
+                    else
+                        std.math.maxInt(u64);
+                    const cr_available = cr_limit - cr.credits_posted - cr.credits_pending;
+                    if (cr_available < amount) amount = cr_available;
+                }
+                // If amount was 0, we will need to return an overflow/exceeds error.
+                break :amount std.math.max(amount, 1);
+            };
```

Comment:
> Also, try to eliminate the overflow/exceeds error earlier, explicitly, rather than using a special case max(1).

Hmm, I'm not quite sure what you're envisioning here.
My rationale for the status quo: The actual error to return is one of `overflows_debits_pending`, `overflows_debits_posted`, `overflows_credits_pending`, `overflows_credits_posted`, `overflows_debits`, `overflows_credits`, `exceeds_credits`, and `exceeds_debits`. 
The `max(1)` trick gives us all of those error checks for free, no need to reimplement them for the at_most flags.